shcomplete/model2correct.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import os
+
+from keras.callbacks import Callback
+from keras.layers import Dense, RepeatVector, Dropout, TimeDistributed, Activation, recurrent
+from keras.layers.embeddings import Embedding
+from keras.models import Sequential, load_model
+from keras.preprocessing.sequence import pad_sequences
+import numpy as np
+from numpy.random import rand, randint, choice
+
+from shcomplete.model2predict import Vocabulary
+
+
+def get_chars(path_to_vocab):
+    """
+    Return the list of all characters found in the vocabulary.
+    """
+    chars = set()
+    with open(path_to_vocab) as f:
+        for line in f:
+            chars.update(line.rstrip())
+    return chars
+
+
+def create_mistakes(chars, command, level_noise):
+    """
+    Add artificial spelling mistakes into a single command.
+    The number of mistakes is proportional to the level of noise we choose.
+    """
+    mistakes = {0: "swap", 1: "deletion", 2: "addition", 3: "transposition"}
```

Comment:
Please make these constants.

```
MISTAKE_SWAP = "swap"
```

etc. Mind the order: the general conecpt first, then the actual value. This allows you to get the full list of constants in IntelliSense instantly.